File: 프로그래머스/lv1/42862. 체육복/체육복.java
import java.util.Set;
import java.util.HashSet;
import java.util.Arrays;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        int answer = n - lost.length;
        Set<Integer> set = new HashSet<> ();
        Arrays.sort(lost);
        
        for(int i=0; i<reserve.length; i++){
            set.add(reserve[i]);
        }
        
        for(int i=0; i<lost.length; i++){
            if(set.contains(lost[i])){
                set.remove(lost[i]);
                lost[i] = -1;
                answer++;
            }
        }
        
        for(int i=0; i<lost.length && set.size()>0; i++){
            if(set.contains(lost[i]-1)){
                answer++;
                set.remove(lost[i]-1);
            }else if(set.contains(lost[i]+1)){
                answer++;
                set.remove(lost[i]+1);
            }
        }
        return answer;
    }
}

File: 프로그래머스/1/42862. 체육복/체육복.js
function solution(n, lost, reserve) {
    let answer = n - lost.length;
    lost.sort((a,b) => a-b);
        
    reserve.forEach(res => {
        if(lost.includes(res)) {
            lost = lost.filter(l => l !== res);
            reserve = reserve.filter(r => r !== res);
            answer++;
        }
    })
        
    for(var i=0; i<lost.length && reserve.length>0; i++){
        if(reserve.includes(lost[i]-1)){
            answer++;
            reserve = reserve.filter(r => r !== lost[i]-1);
        }else if(reserve.includes(lost[i]+1)){
            answer++;
            reserve = reserve.filter(r => r !== lost[i]+1);
        }
    }
    return answer;
}